src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeValueProvider.cs

```diff
@@ -30,8 +30,8 @@ public CompositeValueProvider()
         /// </summary>
         /// <param name="valueProviders">The sequence of <see cref="IValueProvider"/> to add to this instance of
         /// <see cref="CompositeValueProvider"/>.</param>
-        public CompositeValueProvider(IEnumerable<IValueProvider> valueProviders)
-            : base(valueProviders.ToList())
+        protected CompositeValueProvider(IList<IValueProvider> valueProviders)
+            : base(valueProviders)
```

Comment:
Suggest `IReadOnlyList` and sub-classing `ReadOnlyCollection` to express intent a bit better. That is, unless I'm forgetting something that updates the `CompositeValueProvider`.